core/trino-main/pom.xml

```diff
@@ -594,6 +599,41 @@
                             <installDirectory>src/main/resources/webapp/dist</installDirectory>
                         </configuration>
                     </execution>
+                    <execution>
+                        <id>install node and npm (webapp-preview)</id>
+                        <goals>
+                            <goal>install-node-and-npm</goal>
+                        </goals>
+                        <phase>generate-resources</phase>
+                        <configuration>
+                            <workingDirectory>src/main/resources/webapp-preview</workingDirectory>
+                            <installDirectory>src/main/resources/webapp-preview/dist</installDirectory>
```

Comment:
can we install dependencies and build in a single logical step?

Reply:
done. Added a new npm command to do things in one go